pkg/services/accesscontrol/resourcepermissions/store.go

```diff
@@ -671,9 +673,46 @@ func (s *store) createPermissions(sess *db.Session, roleID int64, resource, reso
 		permissions = append(permissions, p)
 	}
 
+	// TODO: Instead of inserting multiple records, we store the actionsset from the actionset service
+	// and then we can check if the action is part of the actionset
+	// check actionSetinMemory
+
+	/*
+		desc:              "should add new resource permission for user",
+		orgID:             1,
+		userID:            1,
+		actions:           []string{"datasources:query", "datasources:write"},
+		resource:          "datasources",
+		resourceID:        "1",
+		resourceAttribute: "uid",
+		seeds: []SetResourcePermissionCommand{
+			{
+				Actions:    []string{"datasources:write"},
+				Resource:   "datasources",
+				ResourceID: "1",
+			},
+		},
+	*/
+	// from actions, scope to ActionSet and a permission
+	// convert actions to []string of actions
+	a := make([]string, 0, len(actions))
+	for action := range actions {
+		a = append(a, action)
+	}
+	actionSetName := s.inmemoryActionSets.ToActionSetName(resource, resourceID, resourceAttribute, a)
+	permissions = append(permissions, accesscontrol.Permission{
```

Comment:
🍖  of the PR, here we add the newly created permission to the set of permissions.

This would later be used as the actual permission check. 

So instead of writing 12 permissions, we now write 13 permissions. 1 for the actionset permission

Reply:
So you're suggesting double-writing the permissions for a bit? Because I think eventually we'd want to switch over to just using action sets and not having to write all the permissions.

Reply:
yea, i'd suggest double-writing. i think this is necessary for backwards compatibility , at least for some versions in-case something happens. 

We might want to split it into 2 db writes, one for the managed permissions one for the action sets. But this would load the db twice.
I opted here to add the action set to the `insertBulk`